dockerfiles/ci/buster/php-8.0/Dockerfile_shared_ext

```diff
@@ -0,0 +1,101 @@
+FROM datadog/dd-trace-ci:buster AS base
+
+ARG phpVersion
+ENV PHP_INSTALL_DIR_DEBUG_ZTS_ASAN=${PHP_INSTALL_DIR}/${phpVersion}-debug-zts-asan
+ENV PHP_INSTALL_DIR_DEBUG_NTS=${PHP_INSTALL_DIR}/${phpVersion}-debug
+ENV PHP_INSTALL_DIR_NTS=${PHP_INSTALL_DIR}/${phpVersion}
+ENV PHP_VERSION=${phpVersion}
+
+FROM base as build
+ARG phpTarGzUrl
+ARG phpSha256Hash
+RUN set -eux; \
+    curl -fsSL -o /tmp/php.tar.gz "${phpTarGzUrl}"; \
+    (echo "${phpSha256Hash} /tmp/php.tar.gz" | sha256sum -); \
+    tar xf /tmp/php.tar.gz -C "${PHP_SRC_DIR}" --strip-components=1; \
+    rm -f /tmp/php.tar.gz; \
+    cd ${PHP_SRC_DIR}; \
+    ./buildconf --force;
+COPY configure_shared_ext.sh /home/circleci/configure.sh
+
+FROM build as php-debug-zts-asan
```

Comment:
Just asking: aren't we testing asan on nts? (I have a :palmface: reaction ready 😄 )

Reply:
No, IIRC we combined ASan and ZTS to hit all the variations in the fewest amount of PHP builds since the container size grows significantly with each PHP build. Right now we have:
- ndbug-nzts
- debug-nzts
- debug-zts-asan

The alternative would have been four builds:
- ndbug-nzts
- debug-nzts
- debug-nzts-asan
- debug-zts

Reply:
I have not  strong opinion. But if the problem is only size I would not consider it to be a problem. If, on the other hand, we feel safe running zts-asan only, I am totally fine with it. I have not enough knowledge to make a call here.

Reply:
Also, even if we decide to do nts+asan, it has not to be done as part f this PR.

Reply:
Ah, thanks for pointing this out. I'm fairly certain that running debug-nzts-asan isn't required if we're running debug-zts-asan, but I see your point that there might be an edge case hiding here.

@morrisonlevi Do you think it is necessary to test debug-nzts-asan in addition to debug-zts-asan? If so, do you think we should rebuild the `datadog/dd-trace-ci:php-8.0_buster` container without ASan and provide a separate container `datadog/dd-trace-ci:php-8.0_buster_asan` with all three builds as ASan builds and test all three? (In a separate PR.)

Reply:
ZTS and non-ZTS may have different code paths sometimes -- so I suppose the correct answer is "yes, ship all the things on ASAN!"

Related, the 2x2 matrix is:

|      | debug | ndebug |
| --- | ----- | ------ |
| zts | [ x]  | [ ] |
| nts | [x] |  [x] |

Should we just bite the bullet and add the missing check here? Theoretically those can have differences too, but less likely so maybe not worth it?

Reply:
In that case, to be as comprehensive as possible, I'm all for building `datadog/dd-trace-ci:php-8.0_buster_asan` and completing the matrix with four ASan builds. #AllTheThings

Reply:
Since we have a card for it, I am resolving this comment for now as the PR is not blocked y this work